app/models/queue_item.rb

```diff
@@ -0,0 +1,19 @@
+class QueueItem < ActiveRecord::Base
+  belongs_to :user 
+  belongs_to :video 
+  
+  delegate :category, to: :video
+  delegate :title, to: :video, prefix: :video 
+  
+  validates_uniqueness_of :video_id, scope: :user_id
+  
+  def rating
+    review = Review.where(user_id: user.id, video_id: video.id).first
```

Comment:
`where(...).first` is not a preferred solution for searching for a single record in Rails 4. `find_by` is.